org.eclipse.lsp4e/src/org/eclipse/lsp4e/MarkerAttributeComputer.java

```diff
@@ -11,89 +11,47 @@
  *******************************************************************************/
 package org.eclipse.lsp4e;
 
-import java.util.HashMap;
 import java.util.Map;
 
-import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
 import org.eclipse.jdt.annotation.NonNull;
 import org.eclipse.jdt.annotation.Nullable;
-import org.eclipse.jface.text.BadLocationException;
 import org.eclipse.jface.text.IDocument;
 import org.eclipse.lsp4j.Diagnostic;
-import org.eclipse.lsp4j.Range;
 
 /**
- * A class that computes the attributes of a
+ * A class that customizes the attributes of a
  * {@link org.eclipse.core.resources.IMarker}.
  *
- * It can be extended by sub-classing it to add additional attributes to the
- * ones already set by this class.
- * <p>
- * When doing so {@link #computeMarkerAttributes(IDocument, Diagnostic)} should
- * reuse the basis implementation and only add attributes from the returned map.
- *
- * Implementations can also customize {@link #attributeCount()} to optimize the
- * size of the attribute map when it is created.
- *
- * <p>
- * Implementation detail: For performance reasons we allow subclasses to add
- * attributes directly to the map returned by
- * {@link #computeMarkerAttributes(IDocument, Diagnostic)}. An attribute
- * containing the language server ID is added later to this map.
- * <p>
  */
 public class MarkerAttributeComputer {
 
-	// Specific marker attributes defined by LSP4E
-	public static final String LSP_DIAGNOSTIC = "lspDiagnostic"; //$NON-NLS-1$
-
 	/**
-	 * Defines the expected number of attributes that will be computed, it is used
-	 * to optimize size of the map at creation time.
+	 * Defines the expected number of attributes that will be added during
+	 * customization, it is used to optimize size of the map at creation time.
 	 *
 	 * @return the expected number of attributes that will be computed
 	 */
 	public int attributeCount() {
-		return 8;
+		return 0;
 	}

```

Comment:
How is this expected to be used in practice?

Reply:
I have removed it, let us keep it simple. Even if we add a bunch of extra attributes, the map should still behave correctly. If I understand correctly, the HashMap implementation will only resize the map when one reaches 64 attributes, and I doubt that will happen.